app/workers/work-failer.ts

```diff
@@ -25,42 +27,50 @@ export default class WorkFailer implements Worker {
   }
 
   /**
-   * Find work items that're older than olderThanMinutes. Fail them and any associated jobs.
-   * @param olderThanMinutes - upper limit on work item age
-   * @param tx - the transaction to use for database interactions
-   * @returns \{ failedWorkItemIds: number[], failedJobIds: string[] \}
+   * Find work items that're older than lastUpdateOlderThanMinutes and call processWorkItemUpdate.
+   * @param lastUpdateOlderThanMinutes - upper limit on the duration since the last update
+   * @returns The ids of items and jobs that were processed: \{ workItemIds: number[], jobIds: string[] \}
    */
-  async failWork(olderThanMinutes: number, tx: Transaction): Promise<{ failedWorkItemIds: number[], failedJobIds: string[] }> {
-    let failedItems: {
-      failedWorkItemIds: number[],
-      failedJobIds: string[]
-    };
-    try {
-      const workItems = await getWorkItemsByAgeAndStatus(
-        tx, olderThanMinutes, [WorkItemStatus.RUNNING],
-      );
-      if (workItems.length) {
-        const workItemIds = workItems.map((item) => item.id);
-        await updateWorkItemStatuses(tx, workItemIds, WorkItemStatus.FAILED);
-        const jobIds = new Set(workItems.map((item) => item.jobID));
-        for (const jobId of jobIds) {
-          const job = await Job.byJobID(tx, jobId, false, true);
-          await completeJob(
-            tx, job, JobStatus.FAILED, this.logger,
-            `Job failed because one or more work items took too long (more than ${olderThanMinutes} minutes) to complete.`,
-          );
+  async processWorkItemUpdates(lastUpdateOlderThanMinutes: number): Promise<{ workItemIds: number[], jobIds: string[] }> {
+    let response: {
+      workItemIds: number[],
+      jobIds: string[]
+    } = { workItemIds: [], jobIds: [] };
+    const workItems = await getWorkItemsByUpdateAgeAndStatus(
+      db, lastUpdateOlderThanMinutes, [WorkItemStatus.RUNNING],
+      [JobStatus.RUNNING, JobStatus.RUNNING_WITH_ERRORS],
+    );
+    if (workItems.length) {
```

Comment:
I think this is taking advantage of 0 being falsy? I'd rewrite it as `workItems.length > 0` or `workItems?.length > 0` if it's possible `getWorkItemsByUpdateAgeAndStatus` doesn't return an empty list when there are no items though it would be best if it returned an empty list.

Reply:
☑️